libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/log/jetty/JettyConnectionLogger.java

```diff
@@ -79,83 +70,21 @@ protected void doStart() {
-                info = ConnectionInfo.from(endpoint);
-                connectionInfo.put(endpoint, info);
-            }
-            if (connection instanceof SslConnection) {
-                SSLEngine sslEngine = ((SslConnection) connection).getSSLEngine();
-                sslToConnectionInfo.put(sslEngine, info);
-            }
-            if (connection.getEndPoint() instanceof ProxyConnectionFactory.ProxyEndPoint) {
-                InetSocketAddress remoteAddress = connection.getEndPoint().getRemoteAddress();
-                info.setRemoteAddress(remoteAddress);
-            }
-        });
-    }
-
-    @Override
-    public void onClosed(Connection connection) {
-        handleListenerInvocation("Connection.Listener", "onClosed", "%h", Stream.of(connection).collect(Collectors.toList()), () -> {
-            SocketChannelEndPoint endpoint = findUnderlyingSocketEndpoint(connection.getEndPoint());
-            ConnectionInfo info = connectionInfo.get(endpoint);
-            if (info == null) {
-                return; // Closed connection already handled
-            }
-            if (!endpoint.isOpen()) {
-                info.setClosedAt(System.currentTimeMillis());
-                // Only print / report failed connections
-                if (info.sslHandshakeFailureException != null) {
-                    connectionLog.log(info.toLogEntry());
-                    metric.increment(METRIC_NAME, info.toMetric());
-                }
-                connectionInfo.remove(endpoint);
-                if (connection instanceof SslConnection) {
-                    SSLEngine sslEngine = ((SslConnection) connection).getSSLEngine();
-                    sslToConnectionInfo.remove(sslEngine);
-                }
-            }
-        });
-    }
-    //
-    // Connection.Listener methods end
-    //
-
-    //
-    // SslHandshakeListener methods start
-    //
-    @Override
-    public void handshakeSucceeded(Event event) {
-        SSLEngine sslEngine = event.getSSLEngine();
-        handleListenerInvocation("SslHandshakeListener", "handshakeSucceeded", "sslEngine=%h", Stream.of(sslEngine).collect(Collectors.toList()), () -> {
-            sslToConnectionInfo.remove(sslEngine);
-        });
-    }
-
     @Override
     public void handshakeFailed(Event event, Throwable failure) {
         SSLEngine sslEngine = event.getSSLEngine();
         handleListenerInvocation("SslHandshakeListener", "handshakeFailed", "sslEngine=%h,failure=%s", Stream.of(sslEngine, failure).collect(Collectors.toList()), () -> {
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug(failure.getMessage());
-            }
-            ConnectionInfo info = sslToConnectionInfo.remove(sslEngine);
+            ConnectionInfo info = ConnectionInfo.from(sslEngine);
             if (info != null) {
                 info.setSslHandshakeFailure(failure);
             }
+
+            if (info.sslHandshakeFailureException != null) {
```

Comment:
in this method I think we're always going to have the exception check so we should able to remove this check.

Reply:
fixed